reco_analysis/reco_analysis/end_detector/end_detector.py

```diff
@@ -0,0 +1,168 @@
+import re
+from langchain_openai import ChatOpenAI
+from langchain_core.prompts import PromptTemplate
+from langchain_core.output_parsers import StrOutputParser
+from dotenv import load_dotenv
+
+# Load environment variables
+load_dotenv("../.env")
+
+# End terms
+END_TERMS = {
+    "goodbye",
+    "bye",
+    "take care",
+    "see you later",
+    "farewell",
+    "talk to you soon",
+    "see you",
+    "later",
```

Comment:
hmm, i might be nitpicking but, I think some of these could be used in a non-goodbye context, and might be safer to remove: take care, later, take it easy.

Reply:
Yeah will remove - agreed.